### PR TITLE
Fix various issues with test_rollout_worker.py when connector is enabled

### PR DESCRIPTION
**Pull request by gjoliver**
Signed-off-by: Jun Gong <jungong@anyscale.com>

## Why are these changes needed?

Fix test_rollout_worker.py

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [*] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/evaluation/tests/test_rollout_worker.py [hunk 1]
```diff
@@ -484,20 +485,31 @@ def step(self, action):
...
             local_worker=ev,
             remote_workers=[],
         )
-        self.assertEqual(max(ev.sample()["rewards"]), 1)
+        self.assertEqual(max(sample["rewards"]), 1)
         result = collect_metrics(ws, [])
-        self.assertEqual(result["episode_reward_mean"], 1000)
+        # Shows different behavior when connector is on/off.
+        if config.enable_connectors:
+            # episode_reward_mean shows the correct clipped value.
```
Comment (ArturNiederfahrenhorst) [inline rllib/evaluation/tests/test_rollout_worker.py:507]: Uh, nice!

File: rllib/evaluation/tests/test_rollout_worker.py [hunk 2]
```diff
@@ -1037,11 +1051,11 @@ def test_wrap_multi_agent_env(self):
     def test_no_training(self):
         class NoTrainingEnv(MockEnv):
             def __init__(self, episode_length, training_enabled):
-                super(NoTrainingEnv, self).__init__(episode_length)
+                super().__init__(episode_length)
                 self.training_enabled = training_enabled
 
             def step(self, action):
-                obs, rew, done, info = super(NoTrainingEnv, self).step(action)
```
Comment (ArturNiederfahrenhorst) [inline rllib/evaluation/tests/test_rollout_worker.py:1044]: This syntax was enabled with Python 3.0. I wonder why it's not part of linters for modern Python projects.

**Review by ArturNiederfahrenhorst — APPROVED** [review]
Great stuff man! Thanks a bunch 👍

File: rllib/evaluation/tests/test_rollout_worker.py (hunk 1 above)
Comment (gjoliver) [reply rllib/evaluation/tests/test_rollout_worker.py:507]: somewhere random man.
as soon as I see this, I was like "let's get this finished and turned on" :)

File: rllib/evaluation/tests/test_rollout_worker.py (hunk 2 above)
Comment (gjoliver) [reply rllib/evaluation/tests/test_rollout_worker.py:1044]: yeah

**Comment by gjoliver** [comment]
looked at all the test failures. they are known master breakages.

**Pull request merged**